pkg/minikube/cluster/cluster.go

```diff
@@ -93,6 +94,18 @@ func StartHost(api libmachine.API, config cfg.MachineConfig) (*host.Host, error)
 		}
 	}
 
+	e := engineOptions(config)
+	if len(e.Env) > 0 {
+		h.HostOptions.EngineOptions.Env = e.Env
+		provisioner, err := provision.DetectProvisioner(h.Driver)
+		if err != nil {
+			return nil, errors.Wrap(err, "Error detecting OS")
```

Comment:
Error messages shouldn't start with capital letters, and it's difficult to determine if the error was actually with OS detection here. How about something like:

errors.Wrap(err, "detecting provisioner")